File: salary.py
import json

data = json.load(open('declarations.json'))['declarations']["declaration"]

for guy in data:
    details = guy["general"]["declarant"]
    nom = "%s %s %s" % (details["civilite"], details["nom"], details["prenom"])
    salarys = guy["mandatElectifDto"].get('items', {})
    if salarys is None:
        salarys = {}
    salarys = salarys.get("items", [])
    if type(salarys) is dict:
        salarys = [salarys]
    print(nom)
    print(len(salarys), 'postes')
    print()
    print()

File: salary.rb
require 'its-it'
require 'json'

data = JSON.load(File.read('declarations.json'))

i = data["declarations"]["declaration"].map do |d|
    b = d["general"]["declarant"]
    name = "#{b["civilite"]} #{b["nom"]} #{b["prenom"]}"
    items = d["mandatElectifDto"].fetch("items", {}).fetch("items", {}) || {}
 
    s = items.map do |t|
        t.fetch("renumeration", {}).fetch("montant", {}).fetch("montant", 0)
    end
    
    [name, s]
end

i = i.sort_by(&:last)

puts i.map(&its.reverse.join(': '))
